src/Components/Modal/Modal.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect } from 'react'
+import Button from '../Button'
+import './Modal.scss'
+
+const Modal = ({
+    show = true, title = 'Modal title', children = 'This is modal content', onSubmit, onClose
+}) => {
+    useEffect(() => {
+        if (show) {
+            document.addEventListener('keydown', logKey)
+        }
+        return () => {
+            document.removeEventListener('keydown', logKey)
+        }
+    }, [show])
+
+    function logKey(e) {
+        if (e.code === 'Escape') {
+            handleClose(onClose)
+        }
+    }
+
+    if (!show) return null
+
+    const handleClose = (action) => {
+        document.querySelector('.modal-dialog').classList.add('modal-dialog--close')
+        document.querySelector('.modal').classList.add('modal--close')
+        setTimeout(() => {
+            action()
+        }, 300)
+    }
+
+    return (
+        <div className="modal fade" tabIndex="-1" role="dialog" aria-labelledby="ModalScrollableTitle" aria-hidden="true" onClick={() => handleClose(onClose)}>
+            <div className="modal-dialog modal-dialog-scrollable" role="document">
```

Comment:
O tai neturi buti modal__dialog, modal__header ir tt. pagal bem?

Reply:
By BEM yes, but here I use clases of Bootstrap with my unique prefix cust-. So any who will use component can take class name from Bootstrap website add suffix and use it in his code

Reply:
There is no effect for BEM stylings.